packages/json-schema/test/integration/build-schema.test.ts

```diff
@@ -0,0 +1,215 @@
+// Copyright IBM Corp. 2013,2018. All Rights Reserved.
+// Node module: @loopback/openapi-spec-builder
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {model, property, ModelMetadataHelper} from '@loopback/repository';
+import {modelToJsonDef, toJsonProperty} from '../../src/build-schema';
+import {expect} from '@loopback/testlab';
+
+describe('build-schema', () => {
+  context('exception', () => {
+    it('errors out when "@property.array" is not used on an array', () => {
+      @model()
+      class BadArray {
+        @property() badArr: string[];
+      }
+      expect(() => {
+        modelToJsonDef(BadArray);
+      }).to.throw(/type is defined as an array/);
+    });
+  });
+  describe('modelToJSON', () => {
+    it('does not convert null or undefined property', () => {
+      @model()
+      class TestModel {
+        @property() nul: null;
+        @property() undef: undefined;
+      }
+      expect(modelToJsonDef(TestModel).properties).to.not.containEql({
```

Comment:
I think it might be better to store the results of the function call and then do the checks. 
```
const jsonDef = modelToJsonDef(TestModel);
expect(jsonDef.properties).to.no.containEql({nul: {type: 'null'}});
expect(modelToJsonDef(TestModel).properties).to.not.containEql({undef: {type: 'undefined'}});
```